pkg/compactor/blocks_cleaner.go

```diff
@@ -493,3 +523,21 @@ func listBlocksOutsideRetentionPeriod(idx *bucketindex.Index, threshold time.Tim
 
 	return
 }
+
+// findLastModifiedTimeForBlock finds the most recent modification time for all files in a block
+func findLastModifiedTimeForBlock(ctx context.Context, blockID ulid.ULID, userBucket objstore.InstrumentedBucket, userLogger log.Logger) (time.Time, error) {
+	var modifiedTime time.Time
+	blockName := blockID.String()
+	err := userBucket.Iter(ctx, blockName, func(name string) error {
+		attrib, err := userBucket.Attributes(ctx, name)
+		if err != nil {
+			level.Warn(userLogger).Log("msg", "couldn't get attribute for file in block", "block", blockName, "file", name)
```

Comment:
If we cannot get attributes for the file, I think we should abort `findLastModifiedTimeForBlock` and not try other files. Better to leave partial block behind than to delete it too early.